src/java.base/share/classes/java/util/ResourceBundle.java

```diff
@@ -1546,7 +1554,22 @@ private static ResourceBundle getBundleImpl(String baseName,
                                                 Locale locale,
                                                 Class<?> caller,
                                                 Control control) {
-        return getBundleImpl(baseName, locale, caller, caller.getClassLoader(), control);
+        ClassLoader loader = getLoader(getCallerModule(caller));
+        return getBundleImpl(baseName, locale, caller, loader, control);
+    }
+
+    /**
+     * Determine the module to be used for the caller.  If
+     * {@link Reflection#getCallerClass()} is called from JNI with an empty
+     * stack frame the caller will be null, so the system class loader unnamed
+     * module will be used.
+     * @param caller
+     * @return
```

Comment:
Maybe you intended to make this `/* .... */` instead of javadoc?  no need to have `@param` and `@return` for this simple method.